pkg/route/secret/monitor_test.go

```diff
@@ -0,0 +1,354 @@
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.FieldSelector = fieldSelector
+			return fakeKubeClient.CoreV1().Secrets(namespace).Watch(ctx, options)
+		},
+	},
+		&corev1.Secret{},
+		0,
+	)
+}
+
+func fakeSecret(namespace, name string) *corev1.Secret {
+	return &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"test": {1, 2, 3, 4},
+		},
+	}
+}
+
+func TestStartInformer(t *testing.T) {
+	scenarios := []struct {
+		name      string
+		isClosed  bool
+		expectErr bool
+	}{
+		{
+			name:      "pass closed channel into informer",
+			isClosed:  true,
+			expectErr: true,
+		},
+		{
+			name:      "pass unclosed channel into informer",
+			isClosed:  false,
+			expectErr: false,
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			fakeKubeClient := fake.NewSimpleClientset()
+			monitor := fakeMonitor(context.TODO(), fakeKubeClient, ObjectKey{})
+			if s.isClosed {
+				close(monitor.stopCh)
+			}
+			go monitor.StartInformer(context.TODO())
+			time.Sleep(10 * time.Millisecond)
+
+			select {
+			// this case will execute if stopCh is closed
+			case <-monitor.stopCh:
+				if !s.expectErr {
+					t.Error("informer is not running")
+				}
+			default:
+				t.Log("informer is running")
+			}
+		})
+	}
+}
+
+func TestStopInformer(t *testing.T) {
+	scenarios := []struct {
+		name             string
+		withStart        bool
+		alreadyStopped   bool
+		expectStopped    bool
+		expectChanClosed bool
+	}{
+		{
+			name:             "stop without starting informer",
+			withStart:        false,
+			expectStopped:    false,
+			expectChanClosed: false,
+		},
+		{
+			name:             "stopping already stopped informer",
+			withStart:        true,
+			alreadyStopped:   true,
+			expectStopped:    false,
+			expectChanClosed: true,
+		},
+		{
+			name:             "correctly stopped informer",
+			withStart:        true,
+			alreadyStopped:   false,
+			expectStopped:    true,
+			expectChanClosed: true,
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			fakeKubeClient := fake.NewSimpleClientset()
+			monitor := fakeMonitor(context.TODO(), fakeKubeClient, ObjectKey{})
+
+			if s.withStart {
+				go monitor.StartInformer(context.TODO())
+				time.Sleep(10 * time.Millisecond)
+			}
+			if s.alreadyStopped {
+				monitor.StopInformer()
+			}
+
+			if monitor.StopInformer() != s.expectStopped {
+				t.Error("unexpected result")
+			}
+
+			var chanClosed bool
+			select {
+			// this case will execute if stopCh is closed
+			case <-monitor.stopCh:
+				chanClosed = true
+			default:
+				chanClosed = false
+			}
+
+			if s.expectChanClosed != chanClosed {
+				t.Error("unexpected result on chan closure")
```

Comment:
```suggestion
				t.Errorf("expceted stop channel closed to be %t but got %t", s.expectedChanClosed, got)
```

Reply:
Done

Reply:
I'm sorry, I typed too fast, another typo: `expceted` -> `expected`.

Reply:
I also didn't observe. Updated it.